ddtrace/contrib/wsgi/wsgi.py

```diff
@@ -129,36 +133,54 @@ def __call__(self, environ, start_response):
                 span_type=SpanTypes.WEB,
             )
 
-            req_span.set_tag_str(COMPONENT, self._config.integration_name)
+            if self.tracer._appsec_enabled:
+                # [IP Blocking]
+                if _context.get_item("http.request.blocked", span=req_span):
+                    start_response("403 FORBIDDEN", [("content-type", headers.get("Accept"))])
+                    closing_iterator = [utils._get_blocked_template(headers.get("Accept")).encode("UTF-8")]
+                    not_blocked = False
+
+            if not_blocked:
+                req_span.set_tag_str(COMPONENT, self._config.integration_name)
+                self._request_span_modifier(req_span, environ)
+                if self.tracer._appsec_enabled:
+                    # [Suspicious Request Blocking on request]
+                    if _context.get_item("http.request.blocked", span=req_span):
+                        start_response("403 FORBIDDEN", [("content-type", headers.get("Accept"))])
+                        closing_iterator = [utils._get_blocked_template(headers.get("Accept")).encode("UTF-8")]
+                        not_blocked = False
```

Comment:
suggestion: We can just return the closing iterator here and avoid creating a response span. This will also simplify the code below. With this change we will need to update the `_TracedIterable` so that it checks if `_self_parent_span is None` before calling `_self_parent_span.finish()`:

```suggestion
                        closing_iterator = [utils._get_blocked_template(headers.get("Accept")).encode("UTF-8")]
                        # We can update the _TracedIterable to only handle a request span.
                        _TracedIterable(iter(closing_iterator), req_span, None)
```